tests/taskgraphs/test_depgraph.py

```diff
@@ -0,0 +1,29 @@
+import unittest
+
+from tiledb.cloud.taskgraphs import depgraph
+
+
+class TestDepGraph(unittest.TestCase):
+    def test_it(self):
+        g = depgraph.DepGraph[str]()
+        g.add_new_node("a", "")
+        g.add_new_node("b", "a")
+        g.add_new_node("c", "b")
+        g.add_new_node("d", "a")
+        g.add_new_node("e", "abcd")
+        self.assertEqual(tuple("abcde"), g.topo_sorted)
```

Comment:
If I read this right:

* node names are required to be single-letter?
* node-lists are multi-letter strings?

that is bound to be limiting

if on the other hand it's the case that some `for e in nodelist:` somewhere happens to work OK whether `nodelist` is like `"abc"` for nodes `"a"`, `"b"`, and `"c"`, _or_ if nodelist is like `["a", "b", "c"]` -- because both string and list-of-string are `Iterable` -- then i think this is overly clever and confusing. if this latter case is true let's please spell out `["a", "b", "c"]` -- even if it takes a few more keystrokes right now. It'll be much less confusing forever after.

Reply:
This isn’t a requirement; it was just me using individual letters as names for the test, and "abcde" (for example) as a sequence of strings. I've expanded these to easier-to-understand names.